### PR TITLE
fix bug transaction history

### PR DESCRIPTION
**Pull request by ltminhthu**

### CODE REVIEW

File: apps/web/src/screens/Profile/components/TransactionHistory/HistoryItem.tsx [hunk 1]
```diff
@@ -92,7 +92,8 @@ const HistoryItem: FC<Transaction> = (transaction) => {
 					numberOfLines={1}
 					ellipsizeMode={'tail'}
 				>
-					{changeBalance} {token.metadata?.symbol ?? 'SOL'}
+					{changeBalance}{' '}
+					{token.metadata?.name?.replaceAll('\u0000', '') ?? 'SOL'}
```
Comment (fluctlight-kayaba) [inline apps/web/src/screens/Profile/components/TransactionHistory/HistoryItem.tsx:96]: don't think default to `SOL` is a good idea - just empty string maybe?

File: apps/web/src/screens/Profile/components/TransactionHistory/TransactionDetails.tsx [hunk 2]
```diff
@@ -67,7 +67,7 @@ const TransactionDetails: FC<Transaction> = ({
 							]}
 						/>
 						<Text style={styles.tokenName}>
-							{amount} {token.metadata?.symbol}
+							{amount} {token.metadata?.name?.replaceAll('\u0000', '').trim()}
```
Comment (fluctlight-kayaba) [inline apps/web/src/screens/Profile/components/TransactionHistory/TransactionDetails.tsx:70]: seems duplicated with `apps/web/src/screens/Profile/components/TransactionHistory/HistoryItem.tsx`, any chance we want a shared util method?

File: packages/engine/solana/transaction.ts [hunk 3]
```diff
@@ -123,17 +123,20 @@ export const getTransactionDetails = async (
...
+		const preBalance = parsedTransaction.meta.preTokenBalances.filter(
+			(item) => item.owner === ownerPublicKey,
+		);
+
+		const postBalance = parsedTransaction.meta.postTokenBalances.filter(
+			(item) => item.owner === ownerPublicKey,
+		);
+
+		if (!preBalance[0].uiTokenAmount.uiAmount) finalTransaction.preBalance = 0;
+		else finalTransaction.preBalance = preBalance[0].uiTokenAmount.uiAmount;
```
Comment (fluctlight-kayaba) [inline packages/engine/solana/transaction.ts:135]:
```typescript
finalTransaction.preBalance = preBalance[0].uiTokenAmount.uiAmount ?
  preBalance[0].uiTokenAmount.uiAmount : 0
```

File: packages/engine/solana/transaction.ts [hunk 4]
```diff
@@ -123,17 +123,20 @@ export const getTransactionDetails = async (
...
 
-		if (!parsedTransaction.meta.postTokenBalances[0].uiTokenAmount.uiAmount)
+		if (!postBalance[0].uiTokenAmount.uiAmount)
 			finalTransaction.postBalance = 0;
-		else
-			finalTransaction.postBalance =
-				parsedTransaction.meta.postTokenBalances[0].uiTokenAmount.uiAmount;
+		else finalTransaction.postBalance = postBalance[0].uiTokenAmount.uiAmount;
```
Comment (fluctlight-kayaba) [inline packages/engine/solana/transaction.ts:139]:
```typescript
finalTransaction.postBalance = postBalance[0].uiTokenAmount.uiAmount ?
  postBalance[0].uiTokenAmount.uiAmount : 0
```

File: apps/web/src/screens/Profile/components/TransactionHistory/TransactionDetails.tsx (hunk 2 above)
Comment (ltminhthu) [reply apps/web/src/screens/Profile/components/TransactionHistory/TransactionDetails.tsx:70]: > seems duplicated with `apps/web/src/screens/Profile/components/TransactionHistory/HistoryItem.tsx`, any chance we want a shared util method?

@cloudle  i got the token name from `getSolanaMetadata`, so i think if we don't want to duplicate this code, we should update `getSolanaMetadata`

**Review by fluctlight-kayaba — APPROVED** [review]

**Pull request merged**